.pipelines/templates/stages/testing_rollback/testing-template.yml

```diff
@@ -0,0 +1,165 @@
+          targetPath: "$(Build.ArtifactStagingDirectory)/"
+        displayName: Download Base Image
+
+      - task: DownloadPipelineArtifact@2
+        inputs:
+          buildType: current
+          artifactName: image-${{ parameters.flavor }}-update-a
+          targetPath: "$(Build.ArtifactStagingDirectory)"
+        displayName: Download Update Image
+
+      - task: DownloadPipelineArtifact@2
+        inputs:
+          buildType: current
+          artifactName: ssh-keys
+          targetPath: "$(Build.ArtifactStagingDirectory)/ssh"
+        displayName: Download SSH Keys
+
+      - task: DownloadPipelineArtifact@2
+        displayName: "Download go-tools"
+        inputs:
+          buildType: current
+          artifactName: "go-tools"
+          patterns: |
+            netlisten
+            storm-trident
+          targetPath: "$(tridentSourceDirectory)/bin"
+
+      - bash: |
+          set -eux
+          chmod +x $(tridentSourceDirectory)/bin/netlisten
+          chmod +x $(tridentSourceDirectory)/bin/storm-trident
+          cp $(Build.ArtifactStagingDirectory)/ssh/id_rsa* ~/.ssh/
+          chmod -R 700 ~/.ssh/
+          mkdir -p $(ob_outputDirectory)
+          ls -l ~/.ssh/
+        displayName: Set up SSH Keys and setup output directory
+
+      - bash: |
+          set -eux
+          ./bin/storm-trident script build-extension-images --build-sysexts --num-clones 3
+          mv ./test-sysext-*.raw $(Build.ArtifactStagingDirectory)/
+
+          find $(Build.ArtifactStagingDirectory)
+        displayName: "Create extensions for test"
+
+      - bash: |
+          set -eux
+          
+          STORM_DYNAMIC_FLAGS=""
+          if [ "${{ parameters.verboseLogging }}" == "True" ]; then
+            STORM_DYNAMIC_FLAGS="$STORM_DYNAMIC_FLAGS --verbose"
+          fi
+          if [ "${{ parameters.flavor }}" != "uki" ]; then
+            STORM_DYNAMIC_FLAGS="$STORM_DYNAMIC_FLAGS --secure-boot"
+          fi
+          if [ "${{ parameters.skipManualRollbackTesting }}" == "true" ]; then
+            # skip manual rollback testing
+            STORM_DYNAMIC_FLAGS="$STORM_DYNAMIC_FLAGS --skip-manual-rollbacks"
+          fi
+          if [ "${{ parameters.skipRuntimeUpdateTesting }}" == "true" ]; then
+            # skip runtime update testing
+            STORM_DYNAMIC_FLAGS="$STORM_DYNAMIC_FLAGS --skip-runtime-updates"
+          fi
+          if [ "${{ parameters.skipExtensionTesting }}" == "true" ]; then
+            # Allow testing for non-extension scenarios
+            STORM_DYNAMIC_FLAGS="$STORM_DYNAMIC_FLAGS --skip-extension-testing"
+          fi
+
+          sudo ./bin/storm-trident run rollback -a $STORM_DYNAMIC_FLAGS \
+            --artifacts-dir $(Build.ArtifactStagingDirectory) \
+            --output-path $(ob_outputDirectory) \
+            --platform ${{ parameters.platform }} \
+            --ssh-private-key-path $(SSH_PRIVATE_KEY_PATH) \
+            --ssh-public-key-path $(SSH_PUBLIC_KEY_PATH) \
```

Comment:
is there a reson to generate these outside of storm instead of inside? 
for the e2e tests i added an RSA generation helper, happy to merge that first if it's useful here